pkg/volume/util/util.go

```diff
@@ -732,3 +734,29 @@ func IsDeviceMountableVolume(volumeSpec *volume.Spec, volumePluginMgr *volume.Vo
 
 	return false
 }
+
+// GetReliableMountRefs calls mounter.GetMountRefs and retries on IsInconsistentReadError.
+// To be used in volume reconstruction of volume plugins that don't have any protection
+// against mounting a single volume on multiple nodes (such as attach/detach).
+func GetReliableMountRefs(mounter mount.Interface, mountPath string) ([]string, error) {
+	var paths []string
+	err := wait.PollImmediate(10*time.Millisecond, time.Minute, func() (bool, error) {
+		var errLogged bool
+		var err error
+		paths, err = mounter.GetMountRefs(mountPath)
+		if io.IsInconsistentReadError(err) {
+			klog.Errorf("Failed to read mount refs from /proc/mounts: %s", mountPath, err)
+			klog.Errorf("The system looks heavily overloaded, will retry indefinitely. Kubelet cannot unmount volumes without getting consistent /proc/mounts")
```

Comment:
"indefinitely" up to a minute?

Reply:
There was endless loop initially, with extra spammy logs. Now that it tries only for one minute it makes sense to log the error only at the end.